src/flb_io_tls.c

```diff
@@ -365,6 +365,10 @@ int net_io_tls_handshake(void *_u_conn, void *_th)
 
         }
 
+        if (u_conn->event.status & MK_EVENT_REGISTERED) {
+            mk_event_del(u->evl, &u_conn->event);
+        }
+
         /*
```

Comment:
I am looking at mk_core code and I see that mk_event_epoll handle mk_event_add() with extra checks compared to mk_event_libevent interface version:

- epoll: https://github.com/fluent/fluent-bit/blob/master/lib/monkey/mk_core/mk_event_epoll.c#L101
- libevent: https://github.com/fluent/fluent-bit/blob/master/lib/monkey/mk_core/mk_event_libevent.c#L102

so in the epoll version, if an event is already registered is assumed that will only change the type or other flag, with this change we incur in two system calls (delete/add) while with checks we only do one change.

On libevent version that check is missing and the fix should be on that file instead of a forced deletion on flb_io_tls.c call.

Reply:
@edsiper Thanks.

Reading the API documentation, libevent seems not to provide an interface
to update an already-registered event. The equivalent of `epoll_ctl()` with
`EPOLL_CTL_MOD` is missing from there.

So it seems like we need to emulate an update using "del/add" calls with libevent.
I implemented it, and just posted it as monkey/monkey/pull/299. I can confirm that
TLS/SSL works fine on Windows with it.

If you notice anything, please feel free to comment!

Reply:
thanks, I've merged Monkey patch 299, would you please adjust this PR too ?

Reply:
Thank you! I updated this PR too.